### ISSUE TITLE
geopy installed but geocoding dialog: install geopy appear...

### ISSUE THREAD

**Issue by marcobra** [elpaso/qgis-geocoding#13]
On Ubuntu 15.04

I already installed geopy (also reinstalled and restarted qgis)

    pip show geopy

    Name: geopy
    Version: 1.10.0
    Location: /usr/local/lib/python2.7/dist-packages/geopy-1.10.0-py2.7.egg

On trying to use plugin i get, dialog menu : Plugins → Geocode → Geocode
```
Couldn't import Python module 'geopy' for communication with geocoders. 
Without it you won't be  able to run GeoCoding plugin. 
You can install 'geopy' with the following command: 'sudo easy_install geopy'.
If you want to access reverse geocoding services, you will need the experimental version, 
more info at ReverseGeocoding
Message: No module named arcgis
```
Hth

**Comment by marcobra** [comment]
After the latest release
Versione installata: 2.10 (in /home/ubuntu/.qgis2/python/plugins/GeoCoding)
Versione disponibile: 2.10 (in Repository plugin ufficiale QGIS)

Still issue on running plugin:
![schermata del 2015-05-06 11 39 41](https://cloud.githubusercontent.com/assets/1607387/7490955/eb0b2a3e-f3e4-11e4-9f96-aecf9c58885f.png)

**Comment by elpaso** [comment]
Cannot reproduce here.
Just to be sure, can you please remove the plugin and install it again? I remember having uploaded the wrong version on the repo and left it there for a few minutes before replacing with the right one, you might have installed the bugged version in that time frame.

BTW the plugin ships with its own version of geopy and should load the internal version and not the system-installed one (if any).

**Comment by marcobra** [comment]
yes reinstalled same issue...

**Comment by marcobra** [comment]
No issue import modules in python in this way:

Python 2.7.9 (default, Apr  2 2015, 15:33:21) 
[GCC 4.9.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from geopy.geocoders.arcgis import ArcGIS
>>> from geopy.geocoders.baidu import Baidu
>>> from geopy.geocoders.bing import Bing
>>> from geopy.geocoders.databc import DataBC
>>> from geopy.geocoders.dot_us import GeocoderDotUS
>>> from geopy.geocoders.geocodefarm import GeocodeFarm
>>> from geopy.geocoders.geonames import GeoNames
>>> from geopy.geocoders.googlev3 import GoogleV3
>>> from geopy.geocoders.opencage import OpenCage
>>> from geopy.geocoders.openmapquest import OpenMapQuest
>>> from geopy.geocoders.navidata import NaviData
>>> from geopy.geocoders.osm import Nominatim
>>> from geopy.geocoders.placefinder import YahooPlaceFinder
>>> from geopy.geocoders.smartystreets import LiveAddress
>>> from geopy.geocoders.what3words import What3Words
>>> from geopy.geocoders.yandex import Yandex
>>> from geopy.geocoders.ignfrance import IGNFrance

**Comment by elpaso** [comment]
I moved geopy in a subfolder, can you please try the version in current master?
You should be able to download it as a zip from this link: https://github.com/elpaso/qgis-geocoding/archive/master.zip

**Comment by marcobra** [comment]
i have update qgis to 2.9.0-Master in my try...

completely removed previous plugin rel. unzipped in .qgis2/python/plugin
enabling plugin menu item (in red color) i get:

impossibile caricare il plugin qgis-geocoding-master a causa di un errore chiamando il metodo classFactory()


Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/qgis/utils.py", line 219, in startPlugin
    plugins[packageName] = package.classFactory(iface)
  File "/home/ubuntu/.qgis2/python/plugins/qgis-geocoding-master/__init__.py", line 24, in classFactory
    from GeoCoding import GeoCoding
  File "/usr/lib/python2.7/dist-packages/qgis/utils.py", line 478, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
  File "/home/ubuntu/.qgis2/python/plugins/qgis-geocoding-master/GeoCoding.py", line 29, in 
    import resources
  File "/usr/lib/python2.7/dist-packages/qgis/utils.py", line 478, in _import
    mod = _builtin_import(name, globals, locals, fromlist, level)
ImportError: No module named resources


Versione Python:
2.7.9 (default, Apr  2 2015, 15:37:21) 
[GCC 4.9.2]


Versione di QGIS:
2.9.0-Master Master, exported

Percorso Python: ['/home/ubuntu/.qgis2/python/plugins/processing', '/usr/share/qgis/python', u'/home/ubuntu/.qgis2/python', u'/home/ubuntu/.qgis2/python/plugins', '/usr/share/qgis/python/plugins', '/usr/local/lib/python2.7/dist-packages/spiderosm-0.4.3-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/pyshp-1.2.1-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/geojson-1.0.9-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/pylev-1.3.0-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/Shapely-1.5.7-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/pyproj-1.9.4-py2.7-linux-x86_64.egg', '/usr/lib/python2.7/dist-packages', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-x86_64-linux-gnu', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages/PILcompat', '/usr/lib/python2.7/dist-packages/gst-0.10', '/usr/lib/python2.7/dist-packages/gtk-2.0', '/usr/lib/pymodules/python2.7', '/usr/lib/python2.7/dist-packages/ubuntu-sso-client', '/usr/lib/python2.7/dist-packages/wx-3.0-gtk2', u'/home/ubuntu/.qgis2//python', '/home/ubuntu/.qgis2/python/plugins/QuickMultiAttributeEdit/forms', '/home/ubuntu/.qgis2/python/plugins/HouseNumbering/forms', '/home/ubuntu/.qgis2/python/plugins/mmqgis/forms', '/usr/share/qgis/python/plugins/fTools/tools', '/home/ubuntu/.qgis2/python/plugins/PointsNumberFromLines/forms']

**Comment by elpaso** [comment]
Oh yes, forgot to mention, you should run "make" in the GeoCoding plugin directory (provided you have installed the toolchain to build PyQt applications) or just copy the following files from the old version:

        Ui_Config.py
        Ui_GeoCoding.py
        Ui_PlaceSelection.py
        resources.py

**Comment by marcobra** [comment]
mv qgis-geocoding-master/ GeoCoding
make
...
ok

Restarted qgis
running plugin Geocode→Geocode


Traceback (most recent call last):
  File "/home/ubuntu/.qgis2/python/plugins/GeoCoding/GeoCoding.py", line 196, in geocode
    geocoder = self.get_geocoder_instance()
  File "/home/ubuntu/.qgis2/python/plugins/GeoCoding/GeoCoding.py", line 258, in get_geocoder_instance
    geocoder = getattr(self.geocoders, geocoder_class)
AttributeError: 'module' object has no attribute 'Nominatim'


Versione Python:
2.7.9 (default, Apr  2 2015, 15:37:21) 
[GCC 4.9.2]


Versione di QGIS:
2.9.0-Master Master, exported

Percorso Python: ['/home/ubuntu/.qgis2/python/plugins/GeoCoding/libs', '/home/ubuntu/.qgis2/python/plugins/processing', '/usr/share/qgis/python', u'/home/ubuntu/.qgis2/python', u'/home/ubuntu/.qgis2/python/plugins', '/usr/share/qgis/python/plugins', '/usr/local/lib/python2.7/dist-packages/spiderosm-0.4.3-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/pyshp-1.2.1-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/geojson-1.0.9-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/pylev-1.3.0-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/Shapely-1.5.7-py2.7.egg', '/usr/local/lib/python2.7/dist-packages/pyproj-1.9.4-py2.7-linux-x86_64.egg', '/usr/lib/python2.7/dist-packages', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-x86_64-linux-gnu', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages/PILcompat', '/usr/lib/python2.7/dist-packages/gst-0.10', '/usr/lib/python2.7/dist-packages/gtk-2.0', '/usr/lib/pymodules/python2.7', '/usr/lib/python2.7/dist-packages/ubuntu-sso-client', '/usr/lib/python2.7/dist-packages/wx-3.0-gtk2', u'/home/ubuntu/.qgis2//python', '/home/ubuntu/.qgis2/python/plugins/QuickMultiAttributeEdit/forms', '/home/ubuntu/.qgis2/python/plugins/HouseNumbering/forms', '/home/ubuntu/.qgis2/python/plugins/mmqgis/forms', '/usr/share/qgis/python/plugins/fTools/tools', '/home/ubuntu/.qgis2/python/plugins/PointsNumberFromLines/forms']

**Comment by marcobra** [comment]
Is right to have /home/ubuntu/.config/ItOpen/GeoCoding.conf  in my sys ?

**Comment by elpaso** [comment]
Yes, they are settings.
I really don't know how to debug this (I cannot reproduce): I guess that it's an import problem with geopy, the idea is that the plugin ships the latest geopy to avoid conflicts with other plugins shipping geopy (we had that in the past). 
To load geopy from the new "libs" dir, we add that lib to the python path: ['/home/ubuntu/.qgis2/python/plugins/GeoCoding/libs',

So, when importing geopy it should be imported from that directory. Can you place a breakpoint before line 258 and check what exactly self.geocoders is? You can inspect __file__, __class__ etc.
It should print something like:

<module 'geopy.geocoders' from '/home/ubuntu/.qgis2/python/plugins/GeoCoding/libs/geopy/geocoders/__init__.pyc'>

Then check if  
print hasattr(self.geocoders, 'Nominatim')
should print True

To make sure it's not a conflict with other plugins, can you try to disable all other plugins?

**Comment by marcobra** [comment]
print hasattr(self.geocoders, 'Nominatim') return false

**Comment by marcobra** [comment]
So must dress or in meanwhile test if there is the geosearch plugin to avoid conflict...

**Comment by elpaso** [comment]
Closed in v 2.11 added an error message when geopy was installed and does not contains the requested geocoder class.
